Vates/VatesAPI/src/vtkMDHexFactory.cpp

```diff
@@ -268,6 +266,44 @@ vtkMDHexFactory::create(ProgressAction &progressUpdating) const {
 }
 
 /*
+ * Get the next highest bin boundary
+ */
+coord_t
+vtkMDHexFactory::getNextBinBoundary(IMDEventWorkspace_sptr imdws) const {
+  auto t_dim = imdws->getTDimension();
+  coord_t bin_width = t_dim->getBinWidth();
+  coord_t dim_min = t_dim->getMinimum();
+  return roundUp(coord_t(m_time) - dim_min, bin_width) + dim_min;
+}
+
+/*
+ * Get the previous bin boundary, or the current one if m_time is on a boundary
+ */
+coord_t
+vtkMDHexFactory::getPreviousBinBoundary(IMDEventWorkspace_sptr imdws) const {
+  auto t_dim = imdws->getTDimension();
+  coord_t bin_width = t_dim->getBinWidth();
+  coord_t dim_min = t_dim->getMinimum();
+  return roundDown(coord_t(m_time) - dim_min, bin_width) + dim_min;
+}
+
+/*
+ * Round up to next multiple of factor
+ * Where "up" means towards +ve infinity
+ */
+coord_t roundUp(const coord_t num_to_round, const coord_t factor) {
+  return (std::floor(num_to_round / factor) + 1) * factor;
+}
+
+/*
+ * Round down to next multiple of factor
+ * Where "down" means towards -ve infinity
+ */
+coord_t roundDown(const coord_t num_to_round, const coord_t factor) {
+  return std::floor(num_to_round / factor) * factor;
+}
+
```

Comment:
The functions are nice. I wonder if these functions can be used anywhere else. We might want to make them furhter available (maybe templated)